packages/shell-api/src/shell-bson.spec.ts

```diff
@@ -0,0 +1,90 @@
+/* eslint @typescript-eslint/camelcase: 0, new-cap: 0, @typescript-eslint/ban-ts-ignore: 0 */
+import shellBson from './shell-bson';
+import { expect } from 'chai';
+
+const hex_1234 = '31323334';
+const b64_1234 = 'MTIzNA==';
+const utf_1234 = '1234';
+
+describe('Shell BSON', () => {
+  describe('Code', () => {
+    it('expects arguments in order', () => {
+      const code = shellBson.Code('code', { k: 'v' });
+      expect(code.code).to.equal('code');
+      expect(code.scope).to.deep.equal({ k: 'v' });
+    });
+  });
+  describe('Date', () => {
+    it('returns string without new', () => {
+      expect(shellBson.Date()).to.be.a('string');
+    });
+    it('returns now object with new', () => {
+      // @ts-ignore
+      const date = new shellBson.Date();
```

Comment:
`any` is usually more kind than ts-ignore: `new (shellBson.Date as any)`. There are more elegant ways to define `Date` as a constructor in `shell-bson`.

Reply:
Unfortunately `as DateConstructor` does not seem to be directly castable
<img width="1439" alt="Screen Shot 2020-05-05 at 14 19 47" src="https://user-images.githubusercontent.com/1045019/81065298-7deeb780-8edb-11ea-842d-96904fc28ac4.png">